setuptools/command/egg_info.py

```diff
@@ -384,10 +384,10 @@ def process_template_line(self, line):
 
         try:
             process_action = action_map[action]
-        except KeyError:
+        except KeyError as e:
             raise DistutilsInternalError(
                 "this cannot happen: invalid action '{action!s}'".format(action=action),
-            )
+            ) from e
```

Comment:
We probably want to improve the error message instead... Maybe something like

```invalid action in `MANIFEST.in`: {action!r}```

... so people actually know where the problem lies.

But other than that I am a bit sceptical that having the info for the KeyError would facilitate the debugging... Also I think it would be nice to preserve the stack trace so `from None` would not work here also...

Reply:
Future reference: see resolution for `raise ... from ...`  in https://github.com/pypa/setuptools/pull/4168#discussion_r1440649502.